ui/app/components/transit-key-actions.js

```diff
@@ -185,7 +187,19 @@ export default Component.extend(TRANSIT_PARAMS, {
     doSubmit(data, options = {}) {
       const { backend, id } = this.getModelInfo();
       const action = this.get('selectedAction');
-      let payload = data ? this.compactData(data) : null;
+      const { encodedBase64, ...formData } = data || {};
+      if (!encodedBase64) {
+        if (action === 'encrypt' && !!formData.plaintext) {
+          formData.plaintext = encodeString(formData.plaintext);
+        }
+        if (action === 'hmac' && !!formData.input) {
```

Comment:
whaddya think about combining the conditional here and on line 198, since they seem to do the same thing?